### PR TITLE
crimson/osd/osd_operations/client_request: Remove seq_mode_t

### PR DESCRIPTION
**Pull request by Matan-B**
`seq_mode_t` is discarded and have no longer in use (Since 5aa342c9f08df251917020d2a0970f94d9d66ec0):
```
if (is_pg_op()) {
  return process_pg_op(pgref);
} else {
  return process_op(
    pgref
  ).then_interruptible([](auto){});
}
```

Signed-off-by: Matan Breizman <mbreizma@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/osd/osd_operations/client_request.cc [hunk 1]
```diff
@@ -221,55 +217,44 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
...
+        }).safe_then_interruptible([] {
+          return PG::load_obc_iertr::make_ready_future<>();
+        });
       }
-      });
-  }).safe_then_interruptible([pg] (const seq_mode_t mode) {
-    return seastar::make_ready_future<seq_mode_t>(mode);
+    });
+  }).safe_then_interruptible([pg] {
+    return seastar::make_ready_future<>();
```
Comment (tchaikov) [inline src/crimson/osd/osd_operations/client_request.cc:249]: how about just replacing `safe_then_interruptible()` with `handle_error_interruptible()`, like

```suggestion
  }).handle_error_interruptible([pg, this] {
    PG::load_obc_ertr::all_same_way([this, pg=std::move(pg)](const auto &code) {
      // ...
  }));  
```

File: src/crimson/osd/osd_operations/client_request.cc [hunk 2]
```diff
@@ -204,12 +201,11 @@ auto ClientRequest::reply_op_error(const Ref<PG>& pg, int err)
   reply->set_reply_versions(eversion_t(), 0);
   reply->set_op_returns(std::vector<pg_log_op_return_item_t>{});
   return conn->send(std::move(reply)).then([] {
-    return seastar::make_ready_future<ClientRequest::seq_mode_t>
-      (seq_mode_t::OUT_OF_ORDER);
+    return seastar::make_ready_future<>();
```
Comment (tchaikov) [inline src/crimson/osd/osd_operations/client_request.cc:204]: could just drop this trailing continuation.

**Review by athanatos — APPROVED** [review]
LGTM with @tchaikov's suggestions

**Review by rzarzynski — APPROVED** [review]

**Comment by Matan-B** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
@Matan-B neither comments was addressed. why did you mark them resolved?

File: src/crimson/osd/osd_operations/client_request.cc (hunk 1 above)
Comment (Matan-B) [reply src/crimson/osd/osd_operations/client_request.cc]: https://github.com/ceph/ceph/pull/49127/files#diff-3465c956b7a8e8fab614e23efef1c6093d8277c66ac36a78a138040010a92eceR246-R250

File: src/crimson/osd/osd_operations/client_request.cc (hunk 2 above)
Comment (Matan-B) [reply src/crimson/osd/osd_operations/client_request.cc]: https://github.com/ceph/ceph/pull/49127/files#diff-3465c956b7a8e8fab614e23efef1c6093d8277c66ac36a78a138040010a92eceR203

**Comment by Matan-B** [comment]
> @Matan-B neither comments was addressed. why did you mark them resolved?

I replied to the comments with the addressed change.

**Comment by tchaikov** [comment]
@Matan-B hi Matan,

```diff
diff --git a/src/crimson/osd/osd_operations/client_request.cc b/src/crimson/osd/osd_operations/client_request.cc
index 4155c5d76cf..6c0dd06ef20 100644
--- a/src/crimson/osd/osd_operations/client_request.cc
+++ b/src/crimson/osd/osd_operations/client_request.cc
@@ -238,8 +238,6 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
               return do_process(ihref, pg, obc);
             });
           });
-        }).safe_then_interruptible([] {
-          return PG::load_obc_iertr::make_ready_future<>();
         });
       }
     });
@@ -274,7 +272,7 @@ ClientRequest::do_process(
     // drop op on the floor; the client will handle returning EIO
     if (m->has_flag(CEPH_OSD_FLAG_SUPPORTSPOOLEIO)) {
       logger().debug("discarding op due to pool EIO flag");
-      return seastar::make_ready_future<seq_mode_t>(seq_mode_t::IN_ORDER);
+      return seastar::now();
     } else {
       logger().debug("replying EIO due to pool EIO flag");
       return reply_op_error(pg, -EIO);
@@ -309,9 +307,7 @@ ClientRequest::do_process(
              ).then_interruptible(
                [this, reply=std::move(reply)]() mutable {
                  logger().debug("{}: sending response", *this);
-                 return conn->send(std::move(reply)).then([] {
-                   return seastar::make_ready_future<>();
-                 });
+                 return conn->send(std::move(reply));
                });
            }, crimson::ct_error::eagain::handle([this, pg, &ihref]() mutable {
              return process_op(ihref, pg);
```

could you check if this patch makes sense?

**Comment by Matan-B** [comment]
> @Matan-B hi Matan,
> 
> ```diff
> diff --git a/src/crimson/osd/osd_operations/client_request.cc b/src/crimson/osd/osd_operations/client_request.cc
> index 4155c5d76cf..6c0dd06ef20 100644
> --- a/src/crimson/osd/osd_operations/client_request.cc
> +++ b/src/crimson/osd/osd_operations/client_request.cc
> @@ -238,8 +238,6 @@ ClientRequest::process_op(instance_handle_t &ihref, Ref<PG> &pg)
>                return do_process(ihref, pg, obc);
>              });
>            });
> -        }).safe_then_interruptible([] {
> -          return PG::load_obc_iertr::make_ready_future<>();
>          });
>        }
>      });
> @@ -274,7 +272,7 @@ ClientRequest::do_process(
>      // drop op on the floor; the client will handle returning EIO
>      if (m->has_flag(CEPH_OSD_FLAG_SUPPORTSPOOLEIO)) {
>        logger().debug("discarding op due to pool EIO flag");
> -      return seastar::make_ready_future<seq_mode_t>(seq_mode_t::IN_ORDER);
> +      return seastar::now();
>      } else {
>        logger().debug("replying EIO due to pool EIO flag");
>        return reply_op_error(pg, -EIO);
> @@ -309,9 +307,7 @@ ClientRequest::do_process(
>               ).then_interruptible(
>                 [this, reply=std::move(reply)]() mutable {
>                   logger().debug("{}: sending response", *this);
> -                 return conn->send(std::move(reply)).then([] {
> -                   return seastar::make_ready_future<>();
> -                 });
> +                 return conn->send(std::move(reply));
>                 });
>             }, crimson::ct_error::eagain::handle([this, pg, &ihref]() mutable {
>               return process_op(ihref, pg);
> ```
> 
> could you check if this patch makes sense?

This is odd, it's only partial section of the commit. I must be missing something.

**Comment by Matan-B** [comment]
> @Matan-B hi Matan,
> could you check if this patch makes sense?

@tchaikov Should be fixed (make check passes as well). Thank you.

**Comment by tchaikov** [comment]
@Matan-B no, they are not fixed. could you double check again?

**Comment by Matan-B** [comment]
> @Matan-B no, they are not fixed. could you double check again?

@tchaikov Sorry did not understand you earlier, fixed now.

**Comment by tchaikov** [comment]
jenkins test submodules

**Comment by tchaikov** [comment]
sorry, i jumped the gun. should have waited until the test passed.

**Pull request merged**